app/components-react/editor/elements/SourceSelector.tsx

```diff
@@ -590,9 +597,18 @@ class SourceSelectorController {
     this.streamingService.actions.setSelectiveRecording(
       !this.streamingService.state.selectiveRecording,
     );
-    if (!this.selectiveRecordingEnabled && this.isDualOutputActive) {
-      this.dualOutputService.actions.toggleDisplay(false, 'vertical');
+    if (this.isDualOutputActive) {
+      this.dualOutputService.actions.toggleDisplay(this.selectiveRecordingEnabled, 'vertical');
```

Comment:
Had a bit of a hard time understanding this one, should we extract a variable even if it's just equal to `selectiveRecordingEnabled` to clarify what we're doing here? i.e `shouldToggleDisplay` or something similar, I'm not sure.